File: GFG/Array/Day4/ReplaceElementsByItsRankInTheArray.java
import java.util.Arrays;
import java.util.HashMap;

public class ReplaceElementsByItsRankInTheArray {
    static int[] replaceWithRank(int arr[], int N) {
        // code here
        int [] temp=new int[N];
        int [] res=new int[N];
        HashMap<Integer,Integer> map=new HashMap<>();
        int index=0;
        
        for(int n:arr){
            temp[index++]=n;
        }
        
        Arrays.sort(temp);
        int rank=1;
        
        for(int n:temp){
            if(!map.containsKey(n)){
                map.put(n,rank++);
            }
        }
        
        index=0;
        
        for(int n:arr){
            res[index++]=map.get(n);
        }
        
        return res;
    }
}

File: GFG/Array/Day4/ReplaceElementsByItsRankInTheArray.py
#User function Template for python3

class Solution:
    def replaceWithRank(self, N, arr):
        # Code here
        arr1=[]
        res=[]
        map={}
        
        for num in arr:
            arr1.append(num)
            
        rank=1
        arr1.sort()
        
        for num in arr1:
            if num not in map:
                map[num]=rank
                rank+=1
            
        for num in arr:
            res.append(map[num])
            
        return res